backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/configuration/IConnectorToolsProviderRegistry.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.diagrams.configuration;
+
+import java.util.Optional;
+
+import org.eclipse.sirius.components.annotations.PublicApi;
+import org.eclipse.sirius.components.collaborative.diagrams.api.IConnectorToolsProvider;
+import org.eclipse.sirius.components.diagrams.description.DiagramDescription;
+
+/**
+ * The registry for connector tools providers.
+ *
+ * @author nvannier
+ */
+@PublicApi
+public interface IConnectorToolsProviderRegistry {
+    void add(IConnectorToolsProvider connectorToolsProvider);
+
+    Optional<IConnectorToolsProvider> get(DiagramDescription representationDescription);
+}
```

Comment:
You don't a "registry" for that, the class `GetConnectorToolsEventHandler` can just be injected with all the `IConnectorToolsProvider`. You can remove this interface.